src/types.ts

```diff
@@ -504,18 +504,59 @@ export interface FluffyStream {
   subtitleType: string;
 }
 
-export interface Metadata {
+export interface Metadata extends partialMetadata {
```

Comment:
Si tu `extends` `PartialMetadata` alors les champs seront optionnels (`?:`) en plus d'être undefinable. Ça ne devrait pas être le cas avec `Metadata`.

Il y a moyen de réutiliser `PartialMetadata` en enlevant les `?:` : 
```suggestion
export type Metadata = { [K in keyof partialMetadata]-?: partialMetadata[K] } & {
```

Reply:
alors ton type est littéralement incompréhensible. Et pour le coup les champs de metadata avec ? sont vraiment optionnels

Reply:
Il sont nullable mais pas optionnel non?  Par optionnel je veux dire que si tu oublis de mettre le field dans l'object typescript ne fera pas de warning et la valeur sera `undefined` (pas `null`).

Reply:
ils sont techniquement les deux. Car flubber va te renvoyer null si l'objet n'existe pas. Mais pour le coup tu peux lui envoyer un json partiel et il est content aussi. J'ai pas hyper envie d'envoyer tout l'objet metadata. Car il est tres rarement tout utilisé